build-ios/build.sh

```diff
@@ -6,7 +6,9 @@ set -e
 
 export BUILD_NAME=official
 export SCONS="scons -j${NUM_CORES} verbose=yes warnings=no progress=no"
-export OPTIONS="debug_symbols=no"
+# Keep LTO disabled for iOS - it works but it makes linking apps on deploy very slow,
+# which is seen as a regression in the current workflow.
+export OPTIONS="production=yes use_lto=no"
```

Comment:
@naithar 3.2.4 RC 2 used this with `OPTIONS="production=yes"` only.
I guess this change should fix https://github.com/godotengine/godot/issues/46064.

Reply:
What about `-flto=thin`, it should be supported and work much faster.

Reply:
We could try it, though that requires changes to `platform/iphone/detect.py`. Might also be worth reworking how we pass LTO options as e.g. for `-flto=thin` on Linux you need to pass `use_lto=yes use_lld=yes use_thinlto=yes`.

Reply:
> What about `-flto=thin`, it should be supported and work much faster.

It was 3x faster for tvOS builds https://github.com/godotengine/godot-build-scripts/pull/21, so we can try using it. I can still provide macOS non-lto builds just in case.

@akien-mga should I also change tvOS build script to support `production` flag?

Reply:
> should I also change tvOS build script to support `production` flag?

Yeah, sounds good.

Reply:
> What about -flto=thin, it should be supported and work much faster.

And it's using much less memory (under 900 MB), waiting for a few minutes for release builds seems pretty reasonable.

Reply:
Also, LTO flags are completely missing from macOS `detect.py`.

Reply:
Yeah I think we should overhaul the LTO flags to make sure they're consistent across all platforms and use the most optimal configuration by default (so `thin` on Clang/LLD).

Reply:
I opened https://github.com/godotengine/godot/issues/46317 to continue the discussion on LTO for iOS and macOS.

Reply:
I had a small doubt whether this override would work... and it turns out it didn't :)
Needs some fixes to how we handle overriding options that would get default values from `dev=yes` and `production=yes`: https://github.com/godotengine/godot/pull/46365